### ISSUE TITLE
deduceCloudEnvironment doesn't add "cloud" in AppEngine

### ISSUE THREAD

**Issue by TWiStErRob** [micronaut-projects/micronaut-core#10799]
### Expected Behavior

As documented:

> The application is running in a Cloud environment (present for all other cloud platform types)
> -- https://docs.micronaut.io/latest/guide/#:~:text=present%20for%20all%20other%20cloud%20platform%20types

### Actual Behaviour

https://github.com/micronaut-projects/micronaut-core/blob/f3c28bee9b14e38ce83f66b9038a4f3e6e2ad85c/inject/src/main/java/io/micronaut/context/env/DefaultEnvironment.java#L742-L747

(compare to branches above, and notice that each add CLOUD explicitly.)

### Steps To Reproduce

I haven't tried, I just stumbled upon the code.

### Environment Information

https://cloud.google.com/appengine/docs/standard/java-gen2/runtime#environment_variables

### Version

4.4.8

**Comment by TWiStErRob** [comment]
Confirmed not working in GAE:

> textPayload: "[2;37m2024-05-04T21:08:54,634 INFO  [main] io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment(DefaultEnvironment.java:168)[m [1;37mEstablished active environments: [gae, gcp, production][m"

**Comment by sdelamo** [comment]
Do you want to submit an PR?

**Comment by TWiStErRob** [comment]
No, because there are no tests for this as far as I found, therefore it should be in scope to add regression tests, which is not as trivial as adding the line I highlighted.

**Comment by pragyamutluru** [comment]
Submitted a PR https://github.com/micronaut-projects/micronaut-core/pull/11035
Tried my best to add some test cases for other scenarios as well.
